wordpress-mobile/WordPress-iOS
# Reader Improvements: Displays Select Interests using FeatureFlag, adds real network calls

Main Issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/14396

WordPressKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/258

- Displays the select interests view if the FeatureFlag is enabled when opening the reader tab
- Updates the select interests view to use the real network calls

### Screenshots
<img src="https://user-images.githubusercontent.com/793774/86956389-daee3000-c10d-11ea-97b9-7628756b478a.gif" width="20%"/>

### To test:
#### 01 - Test feature flag
1. Launch the app
2. Tap on the reader tab
3. Confirm you are not presented the select interests view
4. Tap on the My Site tab
5. Enable the `Reader Improvements Phase 2` feature flag
6. Follow the steps in test 02

#### 02 - Test presenting of the view
Make sure the feature flag is enabled:

1. Launch the app
2. Tap on the Reader tab
3. Wait a second
4. You should be presented with the select interests view

Note: You can dismiss on iOS 13 by dragging the view down

#### 03 - Test interests network
1. Follow the steps from test 02
2. Confirm the interests are displayed in your devices language

#### 04 - Test localization of network interests
1. Change your device language to another language
2. Follow the steps from test 02
3. Confirm the interests are localized to the device language
4. If the language is a right to left language, confirm the interests are aligned to the right

#### 05 - Test slow loading
1. Throttle your internet
1.a alternatively you can add a delay to the `ReaderSelectInterestsViewController refreshData` method as below:

```
    private func refreshData() {
        activityIndicatorView.startAnimating()

        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(1)) {
            self.dataSource.reload()
        }
    }
```
2. Follow the steps from test 02
3. Confirm:
3.a  You see the activity indicator
3.b. The interests load in successfully
3.c The activity indicator is no longer visible


### PR submission checklist:
